Source/JavaScriptCore/dfg/DFGGraph.h

```diff
@@ -1331,13 +1333,12 @@ class Graph final : public virtual Scannable {
             return DontSpeculateInt32;
         
         if (bytecodeCanTruncateInteger(add->arithNodeFlags())) {
-            // If int32 + const double, then we should not speculate this add node with int32 type.
-            // Because ToInt32(int32 + const double) is not always equivalent to int32 + ToInt32(const double).
+            // This function is used in attemptToMakeIntegerAdd. If succeed, both operands will be truncate to integers.
```

Comment:
Nit: "This function is called from attemptToMakeIntegerAdd. If we return SpeculateInt32AndTruncateConstants both operands will be truncated to integers."